src/FixedDataTableContainer.js

```diff
@@ -72,23 +86,44 @@ class FixedDataTableContainer extends React.Component {
   }
 
   render() {
+    const { maxScrollX, scrollX, tableSize } = this.state;
+    const { isRTL } = this.props;
     const fdt = (
-        <FixedDataTable
-            {...this.props}
-            {...this.state}
-            scrollActions={this.scrollActions}
-            columnActions={this.columnActions}
-        />
+      <FixedDataTable
+        {...this.props}
+        {...this.state}
+        scrollActions={this.scrollActions}
+      />
     );
     // For backward compatibility, by default we render FDT-2 scrollbars
     if (this.props.defaultScrollbars) {
       return (
-          <ScrollContainer {...this.props}>
-            {fdt}
-          </ScrollContainer>
+        <PluginContext.Provider
+          value={memoizeContext({
+            maxScrollX,
+            scrollX,
+            isRTL,
+            tableHeight: tableSize.height,
+            touchEnabled: this.props.touchScrollEnabled,
+          })}
+        >
+          <ScrollContainer {...this.props}>{fdt}</ScrollContainer>
+        </PluginContext.Provider>
       );
     }
-    return fdt;
+    return (
+      <PluginContext.Provider
+        value={memoizeContext({
+          maxScrollX,
+          scrollX,
+          isRTL,
+          tableHeight: tableSize.height,
+          touchEnabled: this.props.touchScrollEnabled,
+        })}
+      >
+        {fdt}
+      </PluginContext.Provider>
+    );
```

Comment:
Some props are not needed by other intermediate components and are only needed by plugins. Those can be passed down to plugins through context which will prevent unnecessary rendering of intermediate components due to change in irrelevant props.